autoInstaller.sh

```diff
@@ -335,84 +341,104 @@ if $installgtm || $installYottaDB; then
-
-    # Perform the import
-    su $instance -c "source $basedir/etc/env && $scriptdir/GTM/importVistA.sh"
-
-    # Get GT.M Optimized Routines from Kernel-GTM project and unzip
-    curl -fsSLO --progress-bar https://github.com/shabiel/Kernel-GTM/releases/download/XU-8.0-10001/virgin_install.zip
-
-    # Unzip file, put routines, delete old objects
-    su $instance -c "unzip -qo virgin_install.zip -d $basedir/r/"
-    su $instance -c "unzip -l virgin_install.zip | awk '{print \$4}' | grep '\.m' | sed 's/.m/.o/' | xargs -i rm -fv r/$gtmver/{}"
-    su $instance -c "rm -fv r/$gtmver/_*.o"
-
-    # Get the Auto-configurer for VistA/RPMS and run
-    curl -fsSLO https://raw.githubusercontent.com/shabiel/random-vista-utilities/master/KBANTCLN.m
-    su $instance -c "mv KBANTCLN.m $basedir/r/"
-
-    # Run the auto-configurer accepting the defaults
-    su $instance -c "source $basedir/etc/env && mumps -run START^KBANTCLN"
-
-    # Start Taskman
-    su $instance -c "source $basedir/etc/env && cd ~/tmp/ && mumps -run ^ZTMB"
-elif $installgtm || $installYottaDB; then
-    # Attempt to bypass huge git clone by getting the zip files and unzipping them where they go
-    su $instance -c "source $basedir/etc/env && mkdir -p $basedir/Dashboard"
-    cd $basedir/Dashboard
-    echo "Downloading OSEHRA VistA"
-    curl -fsSL --progress-bar https://github.com/OSEHRA/VistA/archive/master.zip -o VistA-master.zip
-    unzip -q VistA-master.zip
-    rm VistA-master.zip
-    mv VistA-master VistA
-    echo "Downloading OSEHRA VistA-M"
-    curl -fsSL --progress-bar https://github.com/OSEHRA/VistA-M/archive/master.zip -o VistA-M-master.zip
-    unzip -q VistA-M-master.zip
-    rm VistA-M-master.zip
-    mv VistA-M-master VistA-M
-
-    # create random string for build identification
-    # source: http://ubuntuforums.org/showthread.php?t=1775099&p=10901169#post10901169
-    export buildid=`tr -dc "[:alpha:]" < /dev/urandom | head -c 8`
-
-    # Import VistA and run tests using OSEHRA automated testing framework
-    su $instance -c "source $basedir/etc/env && ctest -S $scriptdir/test.cmake -V"
-    # Tell users of their build id
-    echo "Your build id is: $buildid you will need this to identify your build on the VistA dashboard"
-
-    # Compile routines
-    echo "Compiling routines"
-    cd $basedir/r/$gtmver
-    for routine in $basedir/r/*.m; do
-        mumps ${routine} >> $basedir/log/compile.log 2>&1
-    done
-    echo "Done compiling routines"
+if ($installgtm || $installYottaDB); then
+  # Build a dashboard and run the tests to verify installation
+  # These use the Dashboard branch of the VistA repository
+  # The dashboard will clone VistA and VistA-M repos
+  # run this as the $instance user
+  if $generateViVDox; then
+      # unzip a zip of the single .DAT with a directory of routines
+      # and place them in the $basedir 
+      echo "Using local files found in ./GTM/"
+      cd $scriptdir
+      #
+      unzip -q ./GTM/VistA.zip -d /tmp/gtmout
```

Comment:
Where did this file come from?

Reply:
The new VistA.zip is supposed to be supplied by the user, which is described in the OSEHRA documentation for this part of the script.  

That is the basis of it.  From what I think i understood about GDE, you cannot have two regions with "*" as the global name so I had to drop the default file to make our incoming one read by GT.M.  Should I have tried to import them somehow?

Reply:
I'm going to seek some clarification on this as it still seems out of place to me.

1. Adding the zip file of VistA seems to be trying to mimic what the Caché scripts do. The Caché scripts are rather special in that it assumes that you'll get the CACHE.DAT to start. That isn't what I really want the Caché scripts to do in the end, but that is what others who contacted me to do the work wanted. This addition seems to be a new feature rather than using the existing framework (which should require a different flag instead of generateViDox). Why can't you use the pending installation of Routines/Globals to do the work?

2. Can you add a little more documentation to this readme about how this works with GTM/YottaDB. I don't want users to have to keep jumping documentation to figure out how to do things. To me this is a separation of concerns issue: the documentation about how to use the vivian/dox flags to autoInstaller.sh should be documented here. Documentation about vivian/dox in gerneral should be on the osehra pages for vivian/dox.

You are correct that two regions can't have "*" as the global name. I got confused as to what is going on here as this is the section about installing VistA on YottaDB/GTM and wondered why a Vivian/Dox flag is modifying it.

Reply:
If you are adding the feature that one can use a "binary" of the GT.M/YottaDB database it needs to be made clear that the gld needs to come with it.

Reply:
Thanks Christopher,

I'm sorry about the huge delay but I think that I'm ready to push this forward and get this merged in.  
Regarding #1: the point of the script is to allow someone to add their own instance into the ViViaN environment.  I figured the best way to do that consistently would be to expect a "binary" of the MUMPS environment and then put that in when the instance was built in the original script.  Since the Cache instance only has the one install path, it was easy enough to just utilize it again.  I figured GT.M/Yotta would be the same to only install it once but I understand the reasoning behind the separation.  I have since moved the "install" of the local files from the autoInstaller into the vivianInstall script and have now expected the .gld file to come along with the installer.

Regarding #2:  I have added the information from the ViViaN documentation into the README.  

Please take a look at the force-pushed update and let me know if there is anything else that you feel is not up to standards

Reply:
Is looking at this next week ok?

Reply:
Fine by me.